src/condor_utils/file_transfer.cpp

```diff
@@ -4373,6 +4394,29 @@ FileTransfer::computeFileList(
 	ExpandFileTransferList( FilesToSend, filelist, preserveRelativePaths );
 	// dPrintFileTransferList( D_ZKM, filelist, ">>> computeFileList(), after ExpandeFileTransferList():" );
 
+	if (inHandleCommands && m_has_protected_url) {
+		ExprTree * tree = jobAd.Lookup(ATTR_TRANSFER_Q_URL_IN_LIST);
+		if (tree && tree->GetKind() == ClassAd::ExprTree::EXPR_LIST_NODE) {
+			classad::ExprList* list = dynamic_cast<classad::ExprList*>(tree);
+			for(classad::ExprList::iterator it = list->begin() ; it != list->end(); ++it ) {
+				std::string files, attr;
+				classad::Value item;
+				if (jobAd.EvaluateExpr(*it, item, classad::Value::ALL_VALUES) && item.IsStringValue(files)) {
+					if ((*it)->GetKind() == ClassAd::ExprTree::ATTRREF_NODE) {
+						classad::ClassAdUnParser unparser;
+						unparser.SetOldClassAd( true, true );
+						unparser.Unparse(attr, *it);
+						//TODO: De-construct attr for specified xfer queue
+					} else { /*Fail?*/ }
+				} else { /*Fail?*/ }
+				if (files.empty()) { continue; }
+				StringList protectedURLs(files.c_str(), ",");
+				ExpandFileTransferList(&protectedURLs, filelist, preserveRelativePaths, attr.c_str());
```

Comment:
```suggestion
				// We don't have to worry about order in `filelist` because we're going to sort it later.
				ExpandFileTransferList(&protectedURLs, filelist, preserveRelativePaths, attr.c_str());
```